runtime/src/main/jni/Runtime.cpp

```diff
@@ -301,9 +309,39 @@ jint Runtime::GenerateNewObjectId(JNIEnv *env, jobject obj)
 	return objectId;
 }
 
-void Runtime::AdjustAmountOfExternalAllocatedMemoryNative(JNIEnv *env, jobject obj, jlong usedMemory)
+void Runtime::AdjustAmountOfExternalAllocatedMemory() {
+	int64_t usedMemory = m_env.CallLongMethod(m_runtime, GET_USED_MEMORY_METHOD_ID);
+	int64_t changeInBytes = usedMemory - m_lastUsedMemory;
+	int64_t externalMemory = 0;
+
+	if (changeInBytes != 0) {
+		externalMemory = m_isolate->AdjustAmountOfExternalAllocatedMemory(changeInBytes);
+	}
+
+	DEBUG_WRITE("usedMemory=%" PRId64 " changeInBytes=%" PRId64 " externalMemory=%" PRId64, usedMemory, changeInBytes, externalMemory);
+
+	m_lastUsedMemory = usedMemory;
+}
+
+bool Runtime::NotifyGC(JNIEnv *env, jobject obj)
 {
-	m_isolate->AdjustAmountOfExternalAllocatedMemory(usedMemory);
+	bool success = __sync_bool_compare_and_swap(&m_runGC, false, true);
+	return success;
+}
+
+bool Runtime::TryCallGC() {
```

Comment:
Well the global.gc() JS function actually calls this Isolate::RequestGarbageCollectionForTesting() c++ function
It does not seem like something we should be calling in "Release". especially this often as we intent to. The comment on the function state this as well:

> This should only be used for testing purposes and not to enforce a garbage
>    * collection schedule. It has strong negative impact on the garbage
>    * collection performance. Use IdleNotificationDeadline() or
>    * LowMemoryNotification() instead to influence the garbage collection
>    * schedule.
> 

I think we should rethink out approach here.